src/test/java/com/epam/sonarqube/spellcheck/plugin/spellcheck/automaton/states/FinalStateTest.java

```diff
@@ -0,0 +1,73 @@
+package com.epam.sonarqube.spellcheck.plugin.spellcheck.automaton.states;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.*;
+
+import org.mockito.runners.MockitoJUnitRunner;
+
+import com.epam.sonarqube.spellcheck.plugin.spellcheck.automaton.JavaCodeConventionEnglishAutomaton;
+
+@RunWith(MockitoJUnitRunner.class)
+public class FinalStateTest {
+    @Mock
+    private StateCallback callback;
+
+    private FinalState finalState;
+    
+    @Mock
+    private JavaCodeConventionEnglishAutomaton automaton;
+    
+    private StateEvent stateEvent;
+
+    @Before
+    public void init() {
+        stateEvent = new StateEvent(automaton);
+    }
+    
+    @Test
+    public void testCallbackCallsBehavior() throws Exception {
```

Comment:
test should start from "should" and continue as a narrative what it should check
for example shouldVerifyEventPassed and similar